File: frontend/src/api/onboarding.ts
/* 온보딩 API */

import apiClient from './client'

interface OnboardingStatus {
  has_household: boolean
  household_count: number
}

interface CreateHouseholdResponse {
  id: number
  name: string
}

export const onboardingApi = {
  getStatus: () =>
    apiClient.get<OnboardingStatus>('/onboarding/status'),

  createHousehold: (name?: string) =>
    apiClient.post<CreateHouseholdResponse>('/onboarding/create-household', { name }),
}

File: backend/app/schemas/onboarding.py
"""온보딩 관련 스키마"""

from pydantic import BaseModel


class OnboardingStatus(BaseModel):
    """온보딩 상태 응답"""

    has_household: bool
    household_count: int


class CreateDefaultHousehold(BaseModel):
    """기본 가구 생성 요청"""

    name: str | None = None
